src/Umbraco.Core/Services/ContentService.cs

```diff
@@ -2164,7 +2168,12 @@ public IEnumerable<PublishResult> SaveAndPublishBranch(IContent content, bool fo
             // (SaveAndPublishBranchOne does *not* do it)
             scope.Notifications.Publish(
                 new ContentTreeChangeNotification(document, TreeChangeTypes.RefreshBranch, eventMessages));
-            scope.Notifications.Publish(new ContentPublishedNotification(publishedDocuments, eventMessages));
+            if (rootResult?.Success == true)
```

Comment:
Pattern matching is the new hotness 🔥 
```suggestion
            if (rootResult?.Success is true)
```